testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/ResetCredentialsAlternativeFlowsTest.java

```diff
@@ -297,4 +324,110 @@ private void configureResetCredentialsFlowWithoutChooseUser() {
                 .defineAsResetCredentialsFlow()
         );
     }
+
+
+    // Configure reset-credentials flow without 'Send Reset Email' authenticator
+    private void configureResetCredentialsFlowWithoutSendResetEmail(String newFlowAlias) {
+        testingClient.server("test").run(session -> {
+            // Create a copy of the default reset credentials flow with the specified flow alias if it doesn't exist yet
+            if(session.getContext().getRealm().getFlowByAlias(newFlowAlias) == null) {
+                FlowUtil.inCurrentRealm(session).copyResetCredentialsFlow(newFlowAlias);
+            }
+        });
+
+        // Get the ID of 'Send Reset Email' execution within the flow
+        final int sendResetEmailExecutionIndex = realmsResouce().realm("test").flows().getExecutions(newFlowAlias)
```

Comment:
It seems that "configureResetCredentialsFlowWithoutSendResetEmail" and "configureResetCredentialsFlowWithoutResetPassword" is almost same method? Is it possible to create single method instead of those 2 methods and use the label of the authenticator (EG. "Send Reset Email" or "Reset password") as parameters? Bonus point could be to add more generic method to FlowUtil for remove method by providerId (providerId is better than displayName IMO).

Reply:
Right, both of them doing the same, just disabling different authenticator. Ok, will look into FlowUtil method implementation yet (agree, it would be good to have some method universal enough to be re-usable in different places).

Reply:
@iankko Thanks.

Reply:
FWIW regarding this one, tried to refactor the routine to be based in `FlowUtil`, but doesn't seem to be able to get it working right now. When trying it (using either methods of `RealmModel`, `RealmResource`, or `AuthenticationManagementResource` getting either `NotSerializableException` or even something like:

`
Caused by: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [org.keycloak.models.jpa.entities.AuthenticationExecutionEntity#0b77ab43-105f-443f-ab4f-10b96bf38c06]
`
IOW this will need more time to get refactored into the form of a universal `FlowUtil` method. But not to block this PR, for now, just removed the duplication in the methods and kept it under `ResetCredentialsAlternativeFlowsTest.java` location.

This version WFM now:

```
1:22:10,210 INFO  [org.keycloak.testsuite.arquillian.AuthServerTestEnricher] removing test realms after test class
21:22:10,579 INFO  [org.keycloak.testsuite.arquillian.AuthServerTestEnricher] removed realms: test, 
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 46.641 s - in org.keycloak.testsuite.forms.ResetCredentialsAlternativeFlowsTest
21:22:10,648 INFO  [org.keycloak.testsuite.arquillian.undertow.KeycloakOnUndertow] Stopping auth server.
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
```

and can rewrite it into the `FlowUtil` form later (once got it working).